### ISSUE TITLE
behavior of piqi tool and generated ocaml is inconsistent

### ISSUE THREAD

**Issue by eriklovlie** [alavrik/piqi-ocaml#9]
Given this schema:

```
.record [
  .name foo
]

.record [
  .name bar
  .field [ .type int ]
]

.variant [
  .name e
  .option [ .type foo ]
  .option [ .type bar ]
]

.record [
  .name d
  .field [ .name name1 .type e ]
  .field [ .name name2 .type e ]
]

.list [
  .name c
  .type d
]

.record [
  .name thing
  .field [ .name items .type c ]
]

.list [
  .name root
  .type thing
]
```

And this input file:

```
[
[
  .items [
    [
      .foo
      .bar [ 0 ]
    ]
  ]
]
]
```

The following command does not report a problem.

```
piqi check --type path/to/schema/root test.piq
```

But ocaml code generated as follows produces the below exception:

```
piqic-ocaml -C path/to/schema --ext path/to/schema/root.piqi


Uncaught exception:

  Piqirun.Error(-1, "block expected")
```

Placing an empty pair of brackets after `foo` in the input file appears to resolve the problem. I've only just started using piqi, but I would think at least the inconsistent parsing behavior would be an issue worthy of a look.

**Comment by eriklovlie** [comment]
I should say that I'm using `piqic-ocaml` version 0.7.4, installed from opam.

**Comment by alavrik** [comment]
This is a bug. Thanks for reporting it. This input is invalid without extra brackets after `.foo`. `piqi convert` should have failed as well. Give me a day or two and I'll get it fixed. Once fixed the obscure uncaught exception should go away.

**Comment by eriklovlie** [comment]
No problem. I'm not blocked by this so no rush. In fact I think what I really wanted for the `foo` variant option was just a `.option [ .name foo ]`, since there's no data associated with it anyway. That removes the need for an empty list and everything seems fine.

**Comment by alavrik** [comment]
the fix is included in piqilib-0.6.13 and avilable via opam
